x-pack/plugins/ml/public/application/datavisualizer/stats_datagrid/components/field_data_expanded_row/ip_content.tsx

```diff
@@ -9,15 +9,16 @@ import { EuiSpacer } from '@elastic/eui';
 
 import { FormattedMessage } from '@kbn/i18n/react';
 
-import { FieldDataCardProps } from '../field_data_card';
-import { TopValues } from '../top_values';
-import { ExpandedRowFieldHeader } from '../../../../stats_datagrid/components/expanded_row_field_header';
+import type { FieldDataRowProps } from '../../../stats_datagrid/types/field_data_row';
+import { TopValues } from '../../../index_based/components/field_data_row/top_values';
+import { ExpandedRowFieldHeader } from '../expanded_row_field_header';
 
-export const IpContent: FC<FieldDataCardProps> = ({ config }) => {
-  const { stats, fieldFormat } = config;
+export const IpContent: FC<FieldDataRowProps> = ({ config }) => {
+  const { stats } = config;
   if (stats === undefined) return null;
   const { count, sampleCount, cardinality } = stats;
   if (count === undefined || sampleCount === undefined || cardinality === undefined) return null;
+  const fieldFormat = 'fieldFormat' in config ? config.fieldFormat : undefined;
```

Comment:
`const fieldFormat = config.fieldFormat` would be the same here.
Unless this ternary is to show that we're expecting `config.fieldFormat` to be undefined? Otherwise it seems redundant.

Reply:
This is an annoying typescript quirk. This component is now used for both index based and file based data visualizer, and the file based props doesn't have `fieldFormat`. Unfortunately, even if subsequent components understand `fieldFormat` is optional, ts still give the error.
<img width="1043" alt="Screen Shot 2021-01-11 at 16 36 55" src="https://user-images.githubusercontent.com/43350163/104246275-51f0e200-542b-11eb-9980-38fc7ca8da3a.png">